gemini/src/main/java/com/techempower/gemini/path/MethodSegmentHandler.java

```diff
@@ -395,6 +395,9 @@ public boolean handle(PathSegments segments, C context)
   protected PathSegmentMethod getAnnotatedMethod(PathSegments segments, 
       C context) 
   {
+    if (context.getRequestMethod() == null) {
+      return defaultGetMethod;
+    }
```

Comment:
I don't think that this is what we want; if the method is not supported by Gemini then a 501 is probably the most accurate response.